Mathlib/RingTheory/AdicCompletion/Algebra.lean

```diff
@@ -50,20 +50,33 @@ def transitionMapₐ {m n : ℕ} (hmn : m ≤ n) :
     R ⧸ (I ^ n • ⊤ : Ideal R) →ₐ[R] R ⧸ (I ^ m • ⊤ : Ideal R) :=
   AlgHom.ofLinearMap (transitionMap I R hmn) rfl (transitionMap_map_mul I hmn)
 
-/-- `AdicCompletion I R` is an `R`-subalgebra of `∀ n, R ⧸ (I ^ n • ⊤ : Ideal R)`. -/
-def subalgebra : Subalgebra R (∀ n, R ⧸ (I ^ n • ⊤ : Ideal R)) :=
-  Submodule.toSubalgebra (submodule I R) (fun _ ↦ by simp)
-    (fun x y hx hy m n hmn ↦ by simp [hx hmn, hy hmn])
-
-/-- `AdicCompletion I R` is a subring of `∀ n, R ⧸ (I ^ n • ⊤ : Ideal R)`. -/
-def subring : Subring (∀ n, R ⧸ (I ^ n • ⊤ : Ideal R)) :=
-  Subalgebra.toSubring (subalgebra I)
-
-instance : CommRing (AdicCompletion I R) :=
-  inferInstanceAs <| CommRing (subring I)
-
-instance : Algebra R (AdicCompletion I R) :=
-  inferInstanceAs <| Algebra R (subalgebra I)
+instance : Mul (AdicCompletion I R) where
+  mul x y := ⟨x.val * y.val, by simp [x.property, y.property]⟩
+
+instance : One (AdicCompletion I R) where
+  one := ⟨1, by simp⟩
+
+instance : CommRing (AdicCompletion I R) where
```

Comment:
`Function.Injective.commRing` might help

Reply:
I changed it, not entirely sure its better now though.

Reply:
I did another benchmark comparing the version with `Function.Injective.commRing` and the manual one: http://speed.lean-fro.org/mathlib4/compare/04e26275-26be-4d08-8dd1-ffeb4e23bd0e/to/e7b27246-a3e6-496a-b552-ff4b45c7236e?hash2=55cf90705c3509ab5da127b7ecd4db0e147af667
It seems the manual one is significantly faster, so I suggest we leave it at this.